Tests/test_energy.gd

```diff
@@ -0,0 +1,60 @@
+extends TestBase
+
+var card: CardBase
+
+# @overide
+func before_each():
+	super()
+	_player_energy_component.ignore_cost = false
+	_player_energy_component.energy = 0
+	_player_energy_component.energy_generation = 3
+	_player_energy_component.Max_energy = 4
+
+func test_add_energy():
+	card = load("res://Cards/Resource/Card_Energy.tres") # adds 70 energy
+
+	card.on_card_play(_player, [_player])
+	assert_eq(_player_energy_component.energy, 70)
+
+
+func test_using_energy():
+	_player_energy_component.add_energy(10)
+	card = load("res://Cards/Resource/Card_Damage.tres") # costs 7 by default
```

Comment:
does it cost 7 by default or do you want to change it to 7 by default in this test ?
if it's 7 by default, it's too high and it needs to be realistic, you change it to 7 only in this test.